src/SandboxJulia.jl

```diff
@@ -0,0 +1,180 @@
+module SandboxJulia
+
+using LazyArtifacts, Sandbox
+export run_sandboxed_julia
+
+isdebug(group) = Base.CoreLogging.current_logger_for_env(Base.CoreLogging.Debug, group, SandboxJulia) !== nothing
+
+lazy_artifact(x) = @artifact_str(x)
+
+const rootfs_lock = ReentrantLock()
+const rootfs_cache = Dict()
+function prepare_rootfs(distro="debian"; uid=1000, user="sandboxjulia", gid=1000, group="sandboxjulia", home="/home/$user")
+    lock(rootfs_lock) do
+        get!(rootfs_cache, (distro, uid, user, gid, group, home)) do
+            base = lazy_artifact(distro)
+
+            # a bare rootfs isn't usable out-of-the-box
+            derived = mktempdir()
+            cp(base, derived; force=true)
+
+            # add a user and group
+            chmod(joinpath(derived, "etc/passwd"), 0o644)
+            open(joinpath(derived, "etc/passwd"), "a") do io
+                println(io, "$user:x:$uid:$gid::$home:/bin/bash")
+            end
+            chmod(joinpath(derived, "etc/group"), 0o644)
+            open(joinpath(derived, "etc/group"), "a") do io
+                println(io, "$group:x:$gid:")
+            end
+            chmod(joinpath(derived, "etc/shadow"), 0o640)
+            open(joinpath(derived, "etc/shadow"), "a") do io
+                println(io, "$user:*:::::::")
+            end
+
+            # replace resolv.conf
+            rm(joinpath(derived, "etc/resolv.conf"); force=true)
+            write(joinpath(derived, "etc/resolv.conf"), read("/etc/resolv.conf"))
+
+            return (path=derived, uid, user, gid, group, home)
+        end
+    end
+end
+
+"""
+    run_sandboxed_julia(install::String, args=``; env=Dict(), mounts=Dict(),
+                        wait=true, stdin=stdin, stdout=stdout, stderr=stderr,
+                        install_dir="/opt/julia", kwargs...)
+
+Run Julia inside of a sandbox, passing the given arguments `args` to it. The argument `wait`
+determines if the process will be waited on. Streams can be connected using the `stdin`,
+`stdout` and `sterr` arguments. Returns a `Process` object.
+
+Further customization is possible using the `env` arg, to set environment variables, and the
+`mounts` argument to mount additional directories. With `install_dir`, the directory where
+Julia is installed can be chosen.
+"""
+function run_sandboxed_julia(args=``; wait=true,
+                             registries_dir=joinpath(first(DEPOT_PATH), "registries"),
+                             storage_dir=mktempdir(),
+                             install::String=dirname(dirname(Sys.which("julia"))),
```

Comment:
This was a positional argument without a default in PkgEval. Probably I should move it back to that? This was convenient though